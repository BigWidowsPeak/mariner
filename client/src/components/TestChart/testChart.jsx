import React from 'react';
import { LineChart, Line, CartesianGrid, XAxis, YAxis } from 'recharts';

let data = [{'uv': 200, 'Score!': -5}, {'uv': 300, 'Score!': 0}, {'uv': 500, 'Score!': 5}];

const testChart = (props) => (
  <div>
    <LineChart width={400} height={400} data={data}>
      <Line type="monotone" dataKey="uv" stroke="#8884d8" />
      <CartesianGrid stroke="#ccc" />
      <XAxis dataKey="Score!" />
      <YAxis />
    </LineChart>
  </div>
);

export default testChart;