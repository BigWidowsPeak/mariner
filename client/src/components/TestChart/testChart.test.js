import React from 'react';
import TestChart from './testChart.jsx';
import renderer from 'react-test-renderer';

test('Chart renders', () => {
  const component = renderer.create(<TestChart />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});