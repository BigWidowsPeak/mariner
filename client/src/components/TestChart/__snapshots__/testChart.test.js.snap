// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart renders 1`] = `
<div>
  <div
    className="recharts-wrapper"
    style={
      Object {
        "cursor": "default",
        "height": 400,
        "position": "relative",
        "width": 400,
      }
    }
  >
    <svg
      className="recharts-surface"
      height={400}
      style={undefined}
      version="1.1"
      viewBox="0 0 400 400"
      width={400}
    >
      <g
        className="recharts-layer recharts-line"
      >
        <path
          className="recharts-curve recharts-line-curve"
          clipPath={null}
          d="M65,245C120,230,175,215,230,185C285,155,340,110,395,65"
          fill="none"
          height={360}
          stroke="#8884d8"
          strokeDasharray="0px 0px"
          strokeWidth={1}
          width={330}
        />
      </g>
      <g
        className="recharts-cartesian-grid"
      >
        <g
          className="recharts-cartesian-grid-horizontal"
        >
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            width={330}
            x={65}
            x1={65}
            x2={395}
            y={5}
            y1={365}
            y2={365}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            width={330}
            x={65}
            x1={65}
            x2={395}
            y={5}
            y1={275}
            y2={275}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            width={330}
            x={65}
            x1={65}
            x2={395}
            y={5}
            y1={185}
            y2={185}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            width={330}
            x={65}
            x1={65}
            x2={395}
            y={5}
            y1={95}
            y2={95}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            width={330}
            x={65}
            x1={65}
            x2={395}
            y={5}
            y1={5}
            y2={5}
          />
        </g>
        <g
          className="recharts-cartesian-grid-vertical"
        >
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            width={330}
            x={65}
            x1={65}
            x2={65}
            y={5}
            y1={5}
            y2={365}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            width={330}
            x={65}
            x1={230}
            x2={230}
            y={5}
            y1={5}
            y2={365}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            width={330}
            x={65}
            x1={395}
            x2={395}
            y={5}
            y1={5}
            y2={365}
          />
        </g>
      </g>
      <g
        className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
      >
        <line
          className="recharts-cartesian-axis-line"
          fill="none"
          height={30}
          stroke="#666"
          width={330}
          x={65}
          x1={65}
          x2={395}
          y={365}
          y1={365}
          y2={365}
        />
        <g
          className="recharts-cartesian-axis-ticks"
        >
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={330}
              x={65}
              x1={65}
              x2={65}
              y={365}
              y1={371}
              y2={365}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={330}
              x={65}
              y={373}
            >
              <tspan
                dy="0.71em"
                x={65}
              >
                -5
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={330}
              x={65}
              x1={230}
              x2={230}
              y={365}
              y1={371}
              y2={365}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={330}
              x={230}
              y={373}
            >
              <tspan
                dy="0.71em"
                x={230}
              >
                0
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={330}
              x={65}
              x1={395}
              x2={395}
              y={365}
              y1={371}
              y2={365}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={330}
              x={395}
              y={373}
            >
              <tspan
                dy="0.71em"
                x={395}
              >
                5
              </tspan>
            </text>
          </g>
        </g>
      </g>
      <g
        className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
      >
        <line
          className="recharts-cartesian-axis-line"
          fill="none"
          height={360}
          stroke="#666"
          width={60}
          x={5}
          x1={65}
          x2={65}
          y={5}
          y1={5}
          y2={365}
        />
        <g
          className="recharts-cartesian-axis-ticks"
        >
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={360}
              stroke="#666"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={365}
              y2={365}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={360}
              stroke="none"
              textAnchor="end"
              width={60}
              x={57}
              y={365}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                0
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={360}
              stroke="#666"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={275}
              y2={275}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={360}
              stroke="none"
              textAnchor="end"
              width={60}
              x={57}
              y={275}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                150
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={360}
              stroke="#666"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={185}
              y2={185}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={360}
              stroke="none"
              textAnchor="end"
              width={60}
              x={57}
              y={185}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                300
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={360}
              stroke="#666"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={95}
              y2={95}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={360}
              stroke="none"
              textAnchor="end"
              width={60}
              x={57}
              y={95}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                450
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={360}
              stroke="#666"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={5}
              y2={5}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={360}
              stroke="none"
              textAnchor="end"
              width={60}
              x={57}
              y={5}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                600
              </tspan>
            </text>
          </g>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
